### PR TITLE
[ADD] document_page_reference

### PR DESCRIPTION
**Pull request by etobella**
This module allows to add a reference name on documents and simplifies the link
between document pages.

Usage
=====

When editing a document page add elements like ${XXX} where XXX is the reference
of another page. Now, when viewing the document, it will link directly to the page.
Also, the name will be parsed as the display name.

### CODE REVIEW

**Comment by etobella** [comment]
Travis is failing because of #230

File: document_page_reference/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    'author': 'Creu Blanca,Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/knowledge',
+    'depends': [
+        'document_page',
+    ],
+    'data': [
+        'views/document_page.xml',
+        'views/report_document_page.xml',
+    ],
+    'demo': [
```
Comment (jarroyomorales) [inline document_page_reference/__manifest__.py]: Remove if empty

File: document_page_reference/models/document_page.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            return document
+        return False
+
+    def get_reference(self, code):
+        element = self._get_document(code)
+        if not element:
+            return code
+        if self.env.context.get('raw_reference', False):
+            return element.display_name
+        return '<a href="%s">%s</a>' % (
```
Comment (jarroyomorales) [inline document_page_reference/models/document_page.py]: Here you should probably use some regex or external libraries to remove html tags and prevent html injection

**Review by jarroyomorales — COMMENTED** [review]
LGTM! Tested on runbot and it is an awesome feature!

File: document_page_reference/models/document_page.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    reference = fields.Char()
+    content_parsed = fields.Html(compute='_compute_content_parsed')
+
+    @api.depends('history_head')
+    def _compute_content_parsed(self):
+        for record in self:
+            record.content_parsed = record.get_content()
+
+    @api.constrains('reference')
```
Comment (jarroyomorales) [inline document_page_reference/models/document_page.py:59]: It is easier to write it like this
    _sql_constraints = [
        ('reference_uniq', 'unique (reference)', "Reference must be unique !"),
    ]

**Comment by LoisRForgeFlow** [comment]
This is really cool. Just one usability issue I noticed: when you navigate using this reference links you lose the breadcrumb and the menus, see:

no-breadcrums

File: document_page_reference/models/document_page.py (hunk 3 above)
Comment (etobella) [reply document_page_reference/models/document_page.py:59]: Well, reference could be not defined, and also, we must check the format, so I think it is better to maintain the current constrain

File: document_page_reference/models/document_page.py (hunk 2 above)
Comment (etobella) [reply document_page_reference/models/document_page.py]: Comments attended

File: document_page_reference/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Document Page Reference',
+    'summary': """
+        Include references on document pages""",
+    'version': '11.0.1.0.0',
+    'license': 'AGPL-3',
```
Comment (LoisRForgeFlow) [inline document_page_reference/__manifest__.py:9]: I would add here yourself as maintainer and a alpha development status

**Review by LoisRForgeFlow — COMMENTED** [review]
Now looks better, see a suggestion inline

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by etobella** [comment]
Now breadcrumbs is working as expected
:green_apple: :green_apple: :green_apple: :green_apple: 
@jarroyomorales @lreficent

**Review by jarroyomorales — APPROVED** [review]
Nice!

**Review by LoisRForgeFlow — APPROVED** [review]
Great! :clap:

**Review by AdriaGForgeFlow — APPROVED** [review]
Nice work :+1:

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Works like a charm!!

**Pull request merged**